src/vehicle.cpp

```diff
@@ -5393,7 +5393,7 @@ void vehicle::open_or_close(int part_index, bool opening)
       int ydiff = parts[next_index].mount_dy - parts[part_index].mount_dy;
       if((xdiff * xdiff + ydiff * ydiff == 1) && // (x^2 + y^2) == 1
               (part_info(next_index).id == part_info(part_index).id) &&
-              (parts[next_index].open == opening ? 0 : 1)) {
+              (parts[next_index].open == 1 && opening)) {
```

Comment:
> (a == b) ? 0 : 1 <==> (a && b) ? 0 : 1

Ähm, No. `false == false` is true, but `false && false` is false.

How about `.open != opening` without the ternary?

```
     (parts[next_index].open == opening ? 0 : 1)
<==> (parts[next_index].open != opening ? 1 : 0)
<==> (parts[next_index].open != opening ? true : false)
<==> (parts[next_index].open != opening)
```
The only thing is `.open` is an int, not a bool. That might trigger the warning you were about to remove. In that case you might want to warp it: `(.open == 1) != opening`